benchflow-test-manager/application/src/main/java/cloud/benchflow/testmanager/scheduler/TestTaskScheduler.java

```diff
@@ -480,14 +490,19 @@ public void terminateTest(String testID) {
             Future abortRunningTestTaskFuture = taskExecutorService.submit(abortRunningTestTask);
 
             try {
+
               abortRunningTestTaskFuture.get();
+
             } catch (InterruptedException | ExecutionException e) {
               // nothing to do
               e.printStackTrace();
             }
 
           }
 
+          // run the TERMINATING state to terminate
+          handleRunningTest(testID);
```

Comment:
And possibly from another iteration of the outer `while` loop, and not directly here